### ISSUE TITLE
[Bug]: encode_audio reports sample rate as audio_samples

### ISSUE THREAD

**Issue by fedirz** [vllm-project/guidellm#985]
## Bug Description

`encode_audio()` currently populates `audio_samples` using:

```python
"audio_samples": samples.sample_rate
```

This reports the sample rate rather than the number of audio sample frames.

For example, both a one-second and a two-minute 16 kHz recording are reported as having `16,000` samples. These values are later aggregated into benchmark input audio metrics, making sample-count statistics incorrect.

Existing tests primarily use one second of 16 kHz audio, where the sample rate and sample count are coincidentally both `16,000`, masking the defect.

## Expected Behavior

`audio_samples` should contain the number of sample frames represented by the encoded audio.

For same-rate mono audio, a two-second 16 kHz input should report `32,000`, not `16,000`.

If resampling is enabled, the field should be clearly defined and tested as either:

- Decoded input sample frames: `samples.data.shape[-1]`, or
- Encoded/output sample frames, derived from the target sample rate and duration.

Because `audio_bytes` describes the encoded payload, reporting encoded sample frames may be the less surprising interpretation.

## Steps to Reproduce

```python
import torch

from guidellm.utils.audio import encode_audio

result = encode_audio(
    torch.zeros((1, 32000)),
    sample_rate=16000,
    encode_sample_rate=16000,
    audio_format="wav",
)

print(result["audio_seconds"])  # 2.0
print(result["audio_samples"])  # 16000, expected 32000
```

## Operating System

Not OS-specific.

## Python Version

Not Python-version-specific.

## GuideLLM Version

`main` at `39383552962841086d05e25c37b58a83ef06c758`.

## Installation Method

Source checkout.

## Additional Context

Regression coverage should use durations other than exactly one second so the sample rate cannot accidentally equal the expected sample count.
